tests/spec/core/linter-rules/check-punctuation.js

```diff
@@ -0,0 +1,61 @@
+fdescribe("Core Linter Rule - 'check-punctuation'", () => {
+  const ruleName = "check-punctuation";
+  const config = { lint: { [ruleName]: true } };
+  let rule;
+  beforeAll(async () => {
+    rule = await new Promise(resolve => {
+      require([`core/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
+    });
+  });
+  it("checks p ending without a punctuation", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
```

Comment:
Saying "pass" it's a bit confusing here, because this is about catching punctuation errors. This should maybe say:

```JS
should be caught by linter
```  

And variations of the above. 

And: 

```HTML
<p>This is fine.</p>
```

Also, we want to make sure we test a bit thoroughly by adding some whitespace:

```
<p>
  \t Should. be! caught. by? linter 
</p>
```

We also want to check for all the marks:

```
<p>
  This is fine.
</p>
<p>
  This is fine!
</p>
<p>
  This is fine?
</p>
<p>
  This is fine:
</p>
<p>
  !.?: This should be caught by linter
<p/>
```

Reply:
Updated one test case above.

Reply:
Lolol :laughing: The test cases. I'll update them now :+1: